azure-iot-provisioning-devicessdk/azure/iot/provisioning/devicessdk/security/sk_security_provider.py

```diff
@@ -0,0 +1,64 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains a provider that is responsible for providing shared access tokens that will eventually establish
+ the authenticity of devices to Device Provisioning Service.
+"""
+from azure.iot.common.sastoken import SasToken
+
+
+class SymmetricKeySecurityProvider(object):
+    """
+    A provider that is responsible for providing shared access tokens that will eventually establish
+    the authenticity of devices to Device Provisioning Service.
+    """
+
+    def __init__(self, registration_id, symmetric_key, id_scope):
+        """
+        Initialize the symmetric key security client.
+        :param registration_id: The registration ID is used to uniquely identify a device in the Device Provisioning Service.
+        The registration ID is alphanumeric, lowercase string and may contain hyphens.
+        :param symmetric_key: The key which will be used to create the shared access signature token to authenticate
+        the device with the Device Provisioning Service. By default, the Device Provisioning Service creates
+        new symmetric keys with a default length of 32 bytes when new enrollments are saved with the Auto-generate keys
+        option enabled. Users can provide their own symmetric keys for enrollments by disabling this option within
+        16 bytes and 64 bytes and in valid Base64 format.
+        :param id_scope: The ID scope is used to uniquely identify the specific provisioning service the device will
+        register through. The ID scope is assigned to a Device Provisioning Service when it is created by the user and
+        is generated by the service and is immutable, guaranteeing uniqueness.
+        """
```

Comment:
this is VERY good. thanks so much for going through this effort. I would love to see all docs be as precise as this. good work!